sympy/core/add.py

```diff
@@ -593,9 +593,11 @@ def _eval_is_positive(self):
 
     def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
+        from sympy import simplify
+        self = simplify(self)
         if not self.is_number:
             c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
```

Comment:
This code is assuming that `self` is an Add. The above call to simplify breaks that.